src/Sylius/Bundle/ResourceBundle/Controller/ResourceController.php

```diff
@@ -485,7 +485,9 @@ public function applyStateMachineTransitionAction(Request $request): Response
         }
 
         if (!$this->stateMachine->can($configuration, $resource)) {
-            throw new BadRequestHttpException();
```

Comment:
IMO this will break the Admin API. The exception should still be thrown if `!$configuration->isHtmlRequest()` like some lines below to handle a `UpdateHandlingException`: https://github.com/Sylius/Sylius/pull/9355/files#diff-2ea8b7aa90421224fb0d96dd92b4f960L494

WDYT?